stempy/reader.h

```diff
@@ -16,11 +16,21 @@ struct EofException : public std::exception
 struct Header {
   uint32_t imagesInBlock = 0, rows = 0, columns = 0, version = 0, timestamp = 0;
   std::vector<uint32_t> imageNumbers;
+
+  Header() = default;
+  Header(const Header& header) = default;
+  Header(Header&& header) noexcept = default;
 };
 
 struct Block {
   Header header;
-  std::unique_ptr<uint16_t[]> data;
+  std::shared_ptr<uint16_t> data;
```

Comment:
Why shared_ptr over unique_ptr?

Reply:
Memory management gets a little hair when using the Block from the python side, so for now I switch to shared_ptr. When the use-case for the API get a little more define, we may be able to switch back ...